Source/Common/MediaConchLib.h

```diff
@@ -114,6 +114,13 @@ class MediaConchLib
     struct Checker_StatusRes
     {
         Checker_StatusRes() : id(-1), finished(false), has_error(false), percent(NULL), tool(NULL), source_id(-1) {}
+        ~Checker_StatusRes()
+        {
+            if (percent)
+                delete percent;
+            if (tool)
+                delete tool;

```

Comment:
Also set percent and tool to NULL when you delete them.

Reply:
:wave:

Thanks, PR updated.

Reply:
I kindly disagree, as it is in the destructor (the only place not to assign to `NULL`).

Actually, this is the opposite:
- no `if` ([`delete` already does it](https://stackoverflow.com/a/615366/3882655))
- no `NULL` assignment ([it is in the destructor, impossible to reuse the pointer](https://stackoverflow.com/a/3060024/3882655))

Reply:
PR updated